packages/web/src/components/form2/Field.tsx

```diff
@@ -119,6 +119,12 @@ export default function Field({
     }
   }, [rules]);
 
+  useEffect(() => {
+    if (inlineEdit && field?.fieldType === 'response') {
```

Comment:
@saichidvi can you explain this implementation ? why useEffect hook is changed?
why are you setting showDrawer : true when the requirement is to not show drawer?
Also why do you think implementation should be in Field componenet?  How a form should be displayed (overlay or without overlay why will it not be in the formView related component?

Reply:
Can pause for a minute here 
ShowDrawer = true ? What is the meeting of this ? Isn’t the requirement exactly the opposite?- do not show the drawer ?

Reply:
When showDrawer is true we will display the form ... it mean when ever you click on the add formResponse plus icon showDrawer will be set to true and then the form will be displayed to create the response.

Reply:
>  1 . use of showDrawer : true ? 
> 
>   https://github.com/Codemarket2/boossti-frontend/blob/master/packages/web/src/components/form2/Field.tsx#L594
> 
>   If you go though the above link , we can see that we will display the form when the two conditions satisfy .
> 
>  2.  why useEffect hook is changed?
> 
>  I added a new useEffect hook , which is going to display the form when ever you try to add a new value for the field which is of type formResponse.
> 
>  This hook , checks whether we are in inlineEdit state or not if , yes and the field type is also formResponse , then we will set the showDrawer to true such that we will directly display the form . This is to acheive the below function : 
> 
>  
> 
> ![image](https://user-images.githubusercontent.com/91515920/219951713-0d085866-8860-4e42-98bc-7b15b29add6d.png)
> 
> 3. Displaying the form with overlay or without overlay is not related to the formView component . The state or decision of displaying the form on overlay is taken by the component createResponseDrawer here D:\boossti-frontend-new\boossti-frontend\packages\web\src\components\response\CreateResponseDrawer.tsx
@saichidvi What input value to show to use why it should depend on inline edit value yes or no ? And why it should depend on drawer is true or false ?
Isn't what input value to show should just depend on file type ? I see there is check for field type =
from response 
Only two things that matter is what is the object and what action 
Action = create ( since add new value is clicked by user )
And field type = form response and form id can be gotten